dali/test/python/test_operator_sequence_rearrange.py

```diff
@@ -82,22 +82,38 @@ def test_sequence_rearrange():
                 for layout in ["FHW"[:len(shape)], ""]:
                     yield check_sequence_rearrange, 5, shape, new_order, per_sample, dev, layout
 
-@raises(RuntimeError)
 def check_fail_sequence_rearrange(batch_size, shape, reorders, persample_reorder=True, op_type="cpu", layout=""):
     check_sequence_rearrange(batch_size, shape, reorders, persample_reorder, op_type, layout)
 
 def test_fail_sequence_rearrange():
+    shape = [5, 1]
+    orders = [
+        ([6, 7], False),
+        ([-1], False),
+        ([], False),
+        ([np.int32([0]), np.int32([])], True),
+        ([np.int32([6, 7]), np.int32([0])], True),
+        ([np.int32([-1]), np.int32([0])], True),
+        ([np.int32([[1], [2]]), np.int32([[1], [2]])], True)
+    ]
+    error_msgs = [
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'Empty result sequences are not allowed',
+        'Empty `new_order` for sample * is not allowed',
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'new_order[[]*[]] must be between * and input_sequence_length = * for sample *, but it is: *',
+        'Input with dimension * cannot be converted to dimension *'
+    ]
+
     for dev in ["cpu", "gpu"]:
-        for shape in [[5, 1]]:
-            for new_order, per_sample in [([6, 7], False), ([-1], False), ([], False),
-                    ([np.int32([0]), np.int32([])], True), ([np.int32([6, 7]), np.int32([0])], True),
-                    ([np.int32([-1]), np.int32([0])], True), ([np.int32([[1], [2]]), np.int32([[1], [2]])], True)]:
-                yield check_fail_sequence_rearrange, 2, shape, new_order, per_sample, dev
+        for args, error_msg in zip(orders, error_msgs):
```

Comment:
That might be overdoing thigs but I thought about adding assert on equal lengths of both lists before looping over `zip(orders, error_msgs)` just in case somebody adds a case to `orders` in the future and misses the fact that the change should be reflected in error_msgs as zip just happily ignores rest of the longer list.

Reply:
Done